client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -706,6 +834,16 @@ private TxnSegments convert(CreateTxnResponse response) {
                 });
     }
 
+    @Override
+    public void close() {
+        if (closed.compareAndSet(false, true)) {
+            final ScheduledExecutorService executorService = this.internalExecutor.getAndSet(null);
+            if (executorService != null) {
+                executorService.shutdownNow();
```

Comment:
This is only checking and closing  this.internalExecutor. So it is guaranteed to close only internally created executor. Externally supplied executors will not be shutdown